Src/java/engine/src/main/java/org/opencds/cqf/cql/engine/elm/executing/QueryEvaluator.java

```diff
@@ -197,17 +212,27 @@ public static Object internalEvaluate(Query elm, State state, ElmLibraryVisitor<
             }
         }
 
+        if (elm.getAggregate() != null && elm.getReturn() != null) {
+            throw new CqlException("aggregate and return are mutually exclusive");
+        }
+
         if (elm.getReturn() != null && elm.getReturn().isDistinct()) {
             result = DistinctEvaluator.distinct(result, state);
         }
 
+        if (elm.getAggregate() != null) {
+            // TODO: JP - Hmm... this ain't right...
```

Comment:
Do you mean to resolve this TODO before merging?

Reply:
Yes, the implementation is probably wrong. Getting eyes on it so that I can make sure that I understand the correct behavior.